### PR TITLE
Draft

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs [hunk 1]
```diff
@@ -27,7 +27,6 @@ public abstract class DataFlowOperationVisitor<TAnalysisData, TAnalysisContext,
         where TAnalysisContext : AbstractDataFlowAnalysisContext<TAnalysisData, TAnalysisContext, TAnalysisResult, TAbstractAnalysisValue>
         where TAnalysisResult : class, IDataFlowAnalysisResult<TAbstractAnalysisValue>
     {
-#pragma warning disable RS0030 // The symbol 'DiagnosticDescriptor.DiagnosticDescriptor.#ctor' is banned in this project: Use 'DiagnosticDescriptorHelper.Create' instead
```
Comment (Youssef1313) [inline src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs:30]: On my machine, a suggestion (IDE0079 - Remove unnecessary suppression) is reported. However, removing the suppression *should* fail the build. Waiting for the build results here.

Comment (Youssef1313) [reply src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs:30]: Build failed as expected:

> src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs(31,90): error RS0030: (NETCORE_ENGINEERING_TELEMETRY=Build) The symbol 'DiagnosticDescriptor.DiagnosticDescriptor(string, string, string, string, DiagnosticSeverity, bool, string, string, params string[])' is banned in this project: Use 'DiagnosticDescriptorHelper.Create' instead

I'll open a bug for IDE0079 against roslyn.

Comment (mavasani) [reply src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs:30]: Yes, please add it back. The core reason for the unnecessary suppression is: https://github.com/dotnet/roslyn/issues/49925

**Comment by mavasani** [comment]
@Youssef1313 You probably want to close out this PR.

**Comment by Youssef1313** [comment]
Yes, that was only open as a repro for https://github.com/dotnet/roslyn/issues/50558. Closing out.

**Pull request closed without merging**